staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go

```diff
@@ -86,6 +100,10 @@ func (a *mutatingDispatcher) Dispatch(ctx context.Context, attr *generic.Version
 
 // note that callAttrMutatingHook updates attr
 func (a *mutatingDispatcher) callAttrMutatingHook(ctx context.Context, h *v1beta1.Webhook, attr *generic.VersionedAttributes, o admission.ObjectInterfaces) error {
+	reinvokeCtx := attr.GetReinvocationContext()
+	if h.ReinvocationPolicy != nil && *h.ReinvocationPolicy == v1beta1.IfNeededReinvocationPolicy {
+		defer reinvokeCtx.AddReinvocableWebhookToPreviouslyInvoked(h.Name)
```

Comment:
is this defer necessary?

Reply:
`AddReinvocableWebhookToPreviouslyInvoked` does need to be called after the `RequireReinvokingPreviouslyInvokedPlugins` invocation that might be called later on in the function. I'll add a comment explaining.